src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -118,47 +118,62 @@ internal sealed class BuildServerConnection
             var clientDir = buildPaths.ClientDirectory;
             var timeoutNewProcess = timeoutOverride ?? TimeOutMsNewProcess;
             var timeoutExistingProcess = timeoutOverride ?? TimeOutMsExistingProcess;
-            var clientMutexName = GetClientMutexName(pipeName);
             Task<NamedPipeClientStream> pipeTask = null;
-            using (var clientMutex = new Mutex(initiallyOwned: true,
-                                               name: clientMutexName,
-                                               createdNew: out var holdsMutex))
+            Mutex clientMutex = null;
+            var holdsMutex = false;
+            try
             {
                 try
                 {
-                    if (!holdsMutex)
+                    var clientMutexName = GetClientMutexName(pipeName);
```

Comment:
I think this should go outside of the try-catch. This shouldn't throw, and if it does I think we should get a Watson for it.

Reply:
I put it in here because it limits the scope of the variable name better. But I agree it's not part of the catch.

Dislike having to make trade offs like this. Wish we had more granular exception catch support.